usr/lib/cca_stdll/cca_specific.c

```diff
@@ -3310,6 +3323,117 @@ CK_RV token_specific_set_attrs_for_new_object(STDLL_TokData_t *tokdata,
+    long return_code, reason_code, rule_array_count, verb_data_length;
+    unsigned int ver = MAX_INT, rel = MAX_INT, mod = MAX_INT;
+    char ccaversion[8];
+
+    memcpy(rule_array, "STATCCA ", CCA_KEYWORD_SIZE);
+    rule_array_count = 1;
+    verb_data_length = 0;
+    dll_CSUACFQ(&return_code, &reason_code,
+                NULL, NULL,
+                &rule_array_count, rule_array,
+                &verb_data_length, NULL);
+
+    if (return_code != CCA_SUCCESS) {
+        TRACE_ERROR("CSUACFQ (STATCCA) failed. return:%ld, reason:%ld\n",
+                    return_code, reason_code);
+        return CKR_FUNCTION_FAILED;
+    }
+
+    memcpy(ccaversion, &rule_array[CCA_STATCCA_CCA_VERSION_OFFSET],
+           CCA_STATCCA_CCA_VERSION_LENGTH);
+
+    sscanf(ccaversion, "%d.%d.%02d*", (int *)&ver, (int *)&rel, (int *)&mod);
+
+    if (ver < data->min_card_version.ver) {
+        data->min_card_version.ver = ver;
+        data->min_card_version.rel = rel;
+        data->min_card_version.mod = mod;
+        data->card_level_set = 1;
+    } else if (rel < data->min_card_version.rel) {
+        data->min_card_version.rel = rel;
+        data->min_card_version.mod = mod;
+        data->card_level_set = 1;
+    } else if (mod < data->min_card_version.mod) {
+        data->min_card_version.mod = mod;
+        data->card_level_set = 1;
+    }
+
+    return CKR_OK;
+}
+
+/*
+ * Callback function used by cca_get_min_card_level() to determine the
+ * minimum CCA card level among all available APQNs.
+ */
+static CK_RV cca_get_card_level_handler(STDLL_TokData_t *tokdata, const char *adapter,
+                                unsigned short card, unsigned short domain,
+                                void *handler_data)
+{
+    cca_min_card_version_t *data = (cca_min_card_version_t *) handler_data;
+
+    UNUSED(tokdata);
+    UNUSED(adapter);
+    UNUSED(card);
+    UNUSED(domain);
+
+    return cca_get_adapter_version(data);
+}
+
+/* Called during token_specific_init() , no need to obtain CCA adapter lock */
+static CK_RV cca_get_min_card_level(STDLL_TokData_t *tokdata)
+{
+    struct cca_private_data *cca_private = tokdata->private_data;
+    cca_min_card_version_t card_level_data;
+    CK_RV ret;
+
+    /* Determine min card level by iterating over all APQNs */
+    memset(&card_level_data, 0, sizeof(cca_min_card_version_t));
+    card_level_data.min_card_version.ver = MAX_INT;
+    card_level_data.min_card_version.rel = MAX_INT;
+    card_level_data.min_card_version.mod = MAX_INT;
+
+    ret = cca_iterate_adapters(tokdata, cca_get_card_level_handler,
+                               &card_level_data);
+
+    if (ret != CKR_OK || card_level_data.card_level_set == 0) {
+        TRACE_ERROR("cca_iterate_adapters failed, could not determine min card level.\n");
+        ret = CKR_FUNCTION_FAILED;
+        goto done;
+    }
+
+    /* Update min card level in cca_private_data */
+    memcpy(&cca_private->min_card_version, &card_level_data.min_card_version,
```

Comment:
If you use the `struct cca_version` as mentioned above you can do:
`cca_private->min_card_version = card_level_data.min_card_version;`

Reply:
Done.